packages/apple-pay-integration/e2e/ApplePaySessionPaymentStepMockObject.ts

```diff
@@ -0,0 +1,137 @@
+const addApplePaySessionToChrome = () => {
+    class ApplePaySessionPaymentStep implements ApplePaySession {
+        version: number;
+        paymentRequest: ApplePayJS.ApplePayPaymentRequest;
+
+        constructor(version, paymentRequest) {
+            this.version = version;
+            this.paymentRequest = paymentRequest;
+        }
+
+        oncancel: (event: ApplePayJS.Event) => void = () => { 
+            console.log('oncancel');
+        }
+
+        onpaymentauthorized: (event: ApplePayJS.ApplePayPaymentAuthorizedEvent) => void = () => {
+            console.log('onpaymentauthorized');
+        }
+
+        onpaymentmethodselected: (event: ApplePayJS.ApplePayPaymentMethodSelectedEvent) => void = () => {
+            console.log('onpaymentmethodselected');
+        }
+
+        onshippingcontactselected: (event: ApplePayJS.ApplePayShippingContactSelectedEvent) => void = () => { 
+            console.log('onshippingcontactselected');
+        }
+
+        onshippingmethodselected: (event: ApplePayJS.ApplePayShippingMethodSelectedEvent) => void  = () => { 
+            console.log('onshippingmethodselected');
+        }
+
+        onvalidatemerchant: (event: ApplePayJS.ApplePayValidateMerchantEvent) => void  = () => { 
+            console.log('onvalidatemerchant');
+        }
+
+        abort(): void {
+            console.log('abort')
+        }        
+
+        static STATUS_SUCCESS = 1;
+        static STATUS_FAILURE = 2;
+
+        static supportsVersion(_versionNumber) {
+            console.log('supportsVersion', _versionNumber)
+            return true;
+        }
+
+        static canMakePayments() {
+            return true;
+        }
+
+        static canMakePaymentsWithActiveCard() {
+            return Promise.resolve(this.canMakePayments());
+        }
```

Comment:
Think this section can be moved to the top for better orgnization of the class.

Reply:
moved to the top